src/vs/workbench/contrib/terminal/browser/terminalQuickAccess.ts

```diff
@@ -81,7 +81,8 @@ export class TerminalQuickAccessProvider extends PickerQuickAccessProvider<IPick
 
 	private _createPick(terminal: ITerminalInstance, terminalIndex: number, filter: string, groupIndex?: number): IPickerQuickAccessItem | undefined {
 		const iconId = getIconId(terminal);
-		const label = groupIndex ? `$(${iconId}) ${groupIndex + 1}.${terminalIndex + 1}: ${terminal.title}` : `$(${iconId}) ${terminalIndex + 1}: ${terminal.title}`;
+		const terminalExtendedName = terminal.cwd ? `${terminal.title} (${terminal.cwd})` : terminal.title;
```

Comment:
This should use `terminal.description` instead of `.cwd` so they show up exactly as they do in the tab title

Reply:
This is correct depending on what the description setting is set too, we don't want to always show the cwd as we may no actually know it. We want the quick pick to be aligned with what's configured to be displayed in the tabs,

Reply:
I finished it up, thanks for the PR @Balastrong 👍 

![image](https://user-images.githubusercontent.com/2193314/160474811-8dc311d7-b18e-46cb-9a49-0c69bd2e0b36.png)

Reply:
That's great, thank you!